### ISSUE TITLE
AsyncIO support

### ISSUE THREAD

**Issue by satels** [googleapis/google-cloud-python#10104]
I repeat old question: When are you start the supporting AsyncIO up?

The previous official request:
https://github.com/googleapis/google-cloud-python/issues/3103

**Comment by zbarnett** [comment]
I am interested to know this as well.

**Comment by maxpz** [comment]
Really interested in this feature too. Are there some plans on when this is going to be worked on?

**Comment by plamut** [comment]
_Unofficial_ answer - this will not be added until Python 2.7 support is dropped, which will only start happening in the client libraries after the last official Python 2.7.x release (scheduled for April 2020).

After that, the work on AsyncIO support might start, but not necessarily immediately after dropping Python 2.7. The pace will likely differ across various client libraries.

Side note - this issue is in the old monolithic repository, and progress of each individual client library should be tracked in their respective standalone repositories. The links to the new repositories are in the main [README](https://github.com/googleapis/google-cloud-python#google-cloud-python-client) file.

**Comment by whillas** [comment]
Its April. Whats happening?

**Comment by ognacy** [comment]
Any updates on official support for asyncio?

I have ended up re-writing most of my google client code to using REST APIs with aiohttp considering the official client sdk lacks support for it, and some critical components of it lack things like timeout control.

**Comment by robhaswell** [comment]
Any updates?

I have to say, the poor progress of Python client libraries is really making me question using Google Cloud for future projects. AWS has had asyncio support for years.

**Comment by zagfai** [comment]
Waiting for aio support for spanner.. Few months passed.

**Comment by lyricnz** [comment]
Also waiting for this. Frankly, it's a bit embarrassing not to have it already.

**Comment by zagfai** [comment]
We make our lib which support asyncio for spanner open source: https://pypi.org/project/aspanner/

**Comment by parthea** [comment]
Closing as obsolete. The generated clients in this repository have AsyncIO support.

**Comment by irfansharif** [comment]
@parthea, noob question, is there a generated client with AsyncIO support for https://github.com/googleapis/google-cloud-python/tree/main/packages/google-cloud-compute/google/cloud/compute_v1/services/instances? I was unable to find it.

**Comment by parthea** [comment]
@irfansharif You're right. There is no AsyncIO support for `google-cloud-compute`. I'm going to transfer this feature request to the repository for the code generator that we use to generate the `google-cloud-compute` library as I believe the feature needs to be added there.
